numba_dpex/tests/dpjit_tests/dpnp/test_dpnp_empty_like.py

```diff
@@ -46,13 +43,89 @@ def func(a):
 
     assert c.dtype == dtype
     assert c.usm_type == usm_type
-    if device != "unknown":
-        assert (
-            c.sycl_device.filter_string
-            == dpctl.SyclDevice(device).filter_string
-        )
+    assert c.sycl_device.filter_string == device
+    assert c.sycl_queue == dpctl.get_device_cached_queue(device)
+
+
+@pytest.mark.parametrize("shape", shapes)
+@pytest.mark.parametrize("dtype", dtypes)
+@pytest.mark.parametrize("usm_type", usm_types)
+def test_dpnp_empty_like_from_queue(shape, dtype, usm_type):
+    @dpjit
+    def func(a, q):
+        c = dpnp.empty_like(a, dtype=dtype, usm_type=usm_type, sycl_queue=q)
+        return c
+
+    NZ = dpnp.empty(shape)
+    queue = dpctl.SyclQueue()
+
+    try:
+        c = func(NZ, queue)
+    except Exception:
+        pytest.fail("Calling dpnp.empty_like inside dpjit failed")
+
+    if len(c.shape) == 1:
+        assert c.shape[0] == NZ.shape[0]
     else:
-        c.sycl_device.filter_string == dpctl.SyclDevice().filter_string
+        assert c.shape == NZ.shape
+
+    assert c.dtype == dtype
+    assert c.usm_type == usm_type
+    assert c.sycl_queue == NZ.sycl_queue
+    assert c.sycl_queue != queue
+
+
+@pytest.mark.parametrize("shape", shapes)
+@pytest.mark.parametrize("dtype", dtypes)
+@pytest.mark.parametrize("usm_type", usm_types)
+def test_dpnp_empty_like_default(
+    shape,
+    usm_type,
+    dtype,
+):
+    @dpjit
+    def func(arr):
+        c = dpnp.empty_like(arr)
+        return c
+
+    arr = dpnp.empty(shape=shape, usm_type=usm_type, dtype=dtype)
```

Comment:
We do not need to create another array to check if the `dtype`s are the same, we can just check if the returned array comes with the `dtype` that we provided in the argument.